scrapy/resolver.py

```diff
@@ -14,15 +16,23 @@ def __init__(self, reactor, cache_size, timeout):
         self.timeout = timeout
 
     def getHostByName(self, name, timeout=None):
-        if name in dnscache:
-            return defer.succeed(dnscache[name])
+        if PY2 and isinstance(name, text_type):
+            try:
+                safe_name = name.encode('idna')
+            except UnicodeError:
+                # we did our best, let it fail later.
+                safe_name = name
```

Comment:
I would go with replacing `name = name.encode('idna')` instead of assigning to `safe_name` and making this patch larger. Looks like a bug in twisted also, is there a bug report to link to?